File: tools/convert.cpp
#include <lunasvg.h>
#include <iostream>
#include <filesystem>
#include <string>

namespace fs = std::filesystem;
using namespace lunasvg;

void convert_file(const fs::path& src_path, const fs::path& dst_path) {
    auto document = Document::loadFromFile(src_path.string());
    if (!document) {
        std::cerr << "Failed to load: " << src_path << std::endl;
        return;
    }

    auto bitmap = document->renderToBitmap();
    if (!bitmap.isNull()) {
        if (bitmap.writeToPng(dst_path.string())) {
            std::cout << "Converted: " << src_path.filename() << " -> " << dst_path.filename() << std::endl;
        } else {
            std::cerr << "Failed to write PNG: " << dst_path << std::endl;
        }
    }
}

int main(int argc, char* argv[]) {
    if (argc < 3) {
        std::cout << "Usage: convert <input_dir> <output_dir>" << std::endl;
        return 1;
    }

    fs::path src_dir(argv[1]);
    fs::path dst_dir(argv[2]);

    if (!fs::exists(src_dir) || !fs::is_directory(src_dir)) {
        std::cerr << "Error: Input directory does not exist or is not a directory." << std::endl;
        return 1;
    }

    if (!fs::exists(dst_dir)) {
        fs::create_directories(dst_dir);
    }

    for (const auto& entry : fs::directory_iterator(src_dir)) {
        if (entry.is_regular_file() && entry.path().extension() == ".svg") {
            fs::path src_file = entry.path();
            fs::path dst_file = dst_dir / src_file.stem();
            dst_file.replace_extension(".png");

            convert_file(src_file, dst_file);
        }
    }

    std::cout << "Batch conversion complete." << std::endl;
    return 0;
}

File: tools/convert.ts
import { Resvg } from '@resvg/resvg-js';
import fs from 'fs';
import path from 'path';

async function convert(inputDir: string, outputDir: string) {
    if (!fs.existsSync(inputDir)) {
        console.error(`Error: Input directory does not exist: ${inputDir}`);
        process.exit(1);
    }

    if (!fs.existsSync(outputDir)) {
        fs.mkdirSync(outputDir, { recursive: true });
    }

    const files = fs.readdirSync(inputDir);
    for (const file of files) {
        if (file.endsWith('.svg')) {
            const inputPath = path.join(inputDir, file);
            const outputPath = path.join(outputDir, file.replace('.svg', '.png'));

            try {
                const svg = fs.readFileSync(inputPath);
                const resvg = new Resvg(svg);
                const pngData = resvg.render();
                const pngBuffer = pngData.asPng();

                fs.writeFileSync(outputPath, pngBuffer);
                console.log(`Converted: ${file} -> ${path.basename(outputPath)}`);
            } catch (err) {
                console.error(`Failed to convert ${file}:`, err);
            }
        }
    }
    console.log('Batch conversion complete.');
}

const args = process.argv.slice(2);
if (args.length < 2) {
    console.log('Usage: pnpm tsx tools/convert.ts <input_dir> <output_dir>');
    process.exit(1);
}

convert(args[0], args[1]);
